modules/core/src/main/java/org/apache/ignite/internal/util/CollectionUtils.java

```diff
@@ -204,6 +204,68 @@ public boolean contains(Object o) {
         };
     }
 
+    /**
+     * Union lists.
+     *
+     * @param lists Lists.
+     * @param <T> Type of the elements of lists.
+     * @return Immutable union of lists.
+     */
+    @SafeVarargs
+    public static <T> List<T> union(List<T>... lists) {
+        if (lists == null || lists.length == 0) {
+            return List.of();
+        }
+
+        return new AbstractList<>() {
+            /** {@inheritDoc} */
+            @Override
+            public T get(int index) {
+                for (int i = 0; i < lists.length; i++) {
```

Comment:
`for (List<T> list : lists)`?

Reply:
Then more memory will be used.

Reply:
seriously? I think that we agreed that this is not a very good argument

Reply:
What is the best option you suggested?

Reply:
Fix it